sql/parser/sql.y

```diff
@@ -1402,11 +1402,11 @@ rename_stmt:
   }
 | ALTER INDEX qualified_name RENAME TO name
   {
-    $$ = nil
+    $$ = &RenameIndex{Name: $3, NewName: Name($6), IfExists: false}
   }
```

Comment:
@MBerhault Tested on Postgres, seems didn't work!
this is my table:
```
postgres=# \d+ users
                              Table "public.users"
 Column |       Type        | Modifiers | Storage  | Stats target | Description 
--------+-------------------+-----------+----------+--------------+-------------
 id     | integer           | not null  | plain    |              | 
 name   | character varying | not null  | extended |              | 
 title  | character varying |           | extended |              | 
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
    "m" btree (name)
Has OIDs: no
```
this will fail.
```
postgres=# ALTER TABLE users RENAME CONSTRAINT m TO mm;
ERROR:  constraint "m" for table "users" does not exist
```